libstuff/libstuff.cpp

```diff
@@ -62,31 +62,49 @@
 thread_local string SThreadLogPrefix;
 thread_local string SThreadLogName;
 
-// Socket logging variables.
-thread_local struct sockaddr_un SSyslogSocketAddr;
-thread_local int SSyslogSocketFD = 0;
+// We store the process name passed in `SInitialize` to use in logging.
 thread_local string SProcessName;
 
+// This is a set of reusable sockets, each with an associated mutex, to allow for parallel logging directly to the
+// syslog socket, rather than going through the syslog() system call, which goes through journald.
+const size_t S_LOG_SOCKET_MAX = 500;
```

Comment:
Why 500?

Reply:
Arbitrary number that’s more than the number worker threads we use.

Reply:
NAB since the number of worker threads is configurable, should this be either configurable or a N * num_worker_threads?